lib/req/steps.ex

```diff
@@ -36,14 +36,31 @@ defmodule Req.Steps do
   def put_base_url(request)
 
   def put_base_url(%{options: %{base_url: base_url}} = request) do
-    url = URI.merge(base_url, request.url)
-    %{request | url: url}
+    url = to_string(request.url)
+
+    if Regex.match?(~r/^https?:\/\//i, url) do
+      # skip if url is already with scheme
+      request
```

Comment:
Instead of matching the url string, would this work?

```elixir
if request.url.scheme != nil do
```

Reply:
Yes, that seems a lot better.